artifactory_test.go

```diff
@@ -5152,3 +5156,74 @@ func readerCloseAndAssert(t *testing.T, reader *content.ContentReader) {
 func readerGetErrorAndAssert(t *testing.T, reader *content.ContentReader) {
 	assert.NoError(t, reader.GetError(), "Couldn't get reader error")
 }
+
+func TestArtifactoryProjectInitMaven(t *testing.T) {
+	testArtifactoryProjectInit(t, "maven", "multiproject")
+}
+
+func TestArtifactoryProjectInitGradle(t *testing.T) {
+	testArtifactoryProjectInit(t, "gradle", "gradleproject")
+}
+
+func TestArtifactoryProjectInitNpm(t *testing.T) {
+	testArtifactoryProjectInit(t, "npm", "npmproject")
+}
+
+func TestArtifactoryProjectInitGo(t *testing.T) {
+	testArtifactoryProjectInit(t, "go", "dependency")
+}
+
+func TestArtifactoryProjectInitPip(t *testing.T) {
+	testArtifactoryProjectInit(t, "pip", "requirementsproject")
+}
+
+func TestArtifactoryProjectInitNuget(t *testing.T) {
+	testArtifactoryProjectInit(t, "nuget", "multipackagesconfig")
+}
+
+func testArtifactoryProjectInit(t *testing.T, technology, projectExampleName string) {
+	initArtifactoryTest(t)
+	// Create temp JFrog home dir
+	tmpHomeDir, deleteHomeDir := coretests.CreateTempDirWithCallbackAndAssert(t)
+	defer deleteHomeDir()
+	clientTestUtils.SetEnvAndAssert(t, coreutils.HomeDir, tmpHomeDir)
+	_, err := createServerConfigAndReturnPassphrase(t)
+	assert.NoError(t, err)
+
+	// Copy a simple project to a temp work dir
+	tmpWorkDir, deleteWorkDir := coretests.CreateTempDirWithCallbackAndAssert(t)
+	defer deleteWorkDir()
+	testdataSrc := filepath.Join(filepath.FromSlash(tests.GetTestResourcesPath()), technology, projectExampleName)
+	err = fileutils.CopyDir(testdataSrc, tmpWorkDir, true, nil)
+	assert.NoError(t, err)
+
+	// Run cd command to temp dir.
+	currentWd, err := os.Getwd()
+	assert.NoError(t, err)
+	changeDirBack := clientTestUtils.ChangeDirWithCallback(t, currentWd, tmpWorkDir)
+
+	// Run JFrog project init
+	err = platformCli.WithoutCredentials().Exec("project", "init", "--path", tmpWorkDir, "--server-id="+tests.ServerId)
+	assert.NoError(t, err)
+	// Validate correctness of .jfrog/projects/$technology.yml
+	validateProjectYamlFile(t, tmpWorkDir, technology)
+	// Validate correctness of .jfrog/projects/build.yml
+	validateBuildYamlFile(t, tmpWorkDir)
+
+	changeDirBack()
```

Comment:
The ```changeDirBack``` function should probably be called with a defer right after it is received. I found one addition place in this file where this should be fixed as well.